front-end/src/components/analysis-page/IntentLister.jsx

```diff
@@ -63,32 +65,41 @@ class IntentLister extends Component {
         // Ctrl + Right goes to last intent set, defalt Right goes to next intent set
         if (this.state.keysHeld["Control"]) {
             // Set index to that of the first intent in the last intent set
-            this.setState({index: this.props.intents.length - 1 - (this.props.intents.length - 1) % 3});
+            //this.setState({index: this.props.intents.length - 1 - (this.props.intents.length - 1) % 3});
+            store.dispatch(setDisplayingQuestion(this.props.intents.length - 1 - (this.props.intents.length - 1) % 3));
         } else {
             // Increment index by 3 if the index of the last question is beyond the 3 indices currently displayed
-            this.setState({index: Math.min(this.state.index + 3, this.props.intents.length - 1 - (this.props.intents.length - 1) % 3)});
+            const projectId = store.getState().analyzeTranscript.projectIdToBeDisplayed;
+            const indexes = store.getState().analyzeTranscript.DisplayingQuestion;
```

Comment:
@helias22 I want to clarify my understanding on this for when I do intentmenu - so we are storing unique transcript's current index as a dictionary value, and its key is based on project ID? Or something like that. This is a great way to do it!

Reply:
Yup you got it exactly right. It's basically helpful if we have multiple projects but we might not get to that point :((